moto/core/responses.py

```diff
@@ -505,35 +505,35 @@ def _dispatch(self, request: Any, full_url: str, headers: Any) -> TYPE_RESPONSE:
     @staticmethod
     def uri_to_regexp(uri: str) -> str:
         """converts uri w/ placeholder to regexp
-          '/cars/{carName}/drivers/{DriverName}'
-        -> '^/cars/.*/drivers/[^/]*$'
+          '/accounts/{AwsAccountId}/namespaces/{Namespace}/groups'
+        -> '^/accounts/(?P<AwsAccountId>[^/]+)/namespaces/(?P<Namespace>[^/]+)/groups$'
 
-          '/cars/{carName}/drivers/{DriverName}/drive'
-        -> '^/cars/.*/drivers/.*/drive$'
+          '/trustStores/{trustStoreArn+}'
+        -> '^/trustStores/(?P<trustStoreArn>.+)$'
 
         """
 
-        def _convert(elem: str, is_last: bool) -> str:
+        def _convert(elem: str) -> str:
             if not re.match("^{.*}$", elem):
                 # URL-parts sometimes contain a $
                 # Like Greengrass: /../deployments/$reset
                 # We don't want to our regex to think this marks an end-of-line, so let's escape it
                 return elem.replace("$", r"\$")
+
+            # When the element ends with +} the parameter can contain a / otherwise not.
+            no_slash_allowed = False if "+" == elem[-2:-1] else True
```

Comment:
Any objections to making the naming positive, and having a more direct check? Something like this is more readable IMO:
```
slash_allowed = elem.endswith("+}")
if slash_allowed:
    return ...
return ...
```